src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AbstractRazorDelegatingEndpoint.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using OmniSharp.Extensions.JsonRpc;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal abstract class AbstractRazorDelegatingEndpoint<TRequest, TResponse, TDelegatedParams> : IJsonRpcRequestHandler<TRequest, TResponse?>
```

Comment:
I guess I'm kind of fuzzy on why we need this class over doing [what we do for DocumentColor](https://github.com/dotnet/razor-tooling/blob/18c929f647abbd204dd81378ab727adcb7109dd3/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentColor/DocumentColorEndpoint.cs#L35). Probably that just means I'm not understanding the core goals here but could you expand a bit?

Reply:
It's standardizing a few checks, one of which is the check for single server being enabled. The hope is to reduce code duplication here. DocumentColor seems simple enough, but other pieces were rewriting a lot of the same logic. Ideally with this base it will be simple to convert other features to single server.

Reply:
Oh nice, didn't know about document color, but its definitely in the single server ethos. I think in future once we remove the single server feature flag and roll it out, we could definitely follow up and I think the document color endpoint might end up being little to no code, if it uses this infra.

Reply:
Yea. I also think folding range will be very little code as well. It will be interesting once we get rid of the flag what features can shift. I think everything in the `RazorCustomMessageTarget` should be able to use this.